app/models/minion.rb

```diff
@@ -54,6 +63,18 @@ def self.assign_roles!(roles: {}, default_role: :minion)
     end
   end
 
+  def self.select_masters(roles)
```

Comment:
We have been using keyword args, I think it would be good to stick with them for easier extensibility.